puzzle.py

```diff
@@ -0,0 +1,199 @@
+
+from trie import Trie
+import random
+
+class Puzzle:
+	"""
+	Create a Puzzle and solve for all the words in the puzzle
+	"""
+	def __init__(self, size=None, words_dictionary=None, puzzle_matrix=None):
+		"""
+		1. initalizes each cell with a letter at random
+		2. reads the dictionary for all words
+		3. constructs puzzle matrix
+
+		"""
+		
+		#initialize the trie data structure
+		self.__trie = Trie();
```

Comment:
double underscores are somewhat discouraged in python, to avoid unintentionally overriding python internals.

Reply:
I read about it a bit did not get proper documentation on what is preferred and decided to use __ . I will update the code to use _ (single) for private methods.

Reply:
This was the doc page that I referred to https://docs.python.org/3/tutorial/classes.html#tut-private before using the __ syntax.